### PR TITLE
add support for tokio

### PR DESCRIPTION
**Pull request by joriskleiber**
## Goal
Add support for tokio's wrapper types

## Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/Aleph-Alpha/ts-rs/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: ts-rs/Cargo.toml [hunk 1]
```diff
@@ -35,11 +35,13 @@ smol_str-impl = ["smol_str"]
 serde-json-impl = ["serde_json"]
 no-serde-warnings = ["ts-rs-macros/no-serde-warnings"]
 import-esm = []
+tokio-impl = ["tokio"]
 
 [dev-dependencies]
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1"
 chrono = { version = "0.4", features = ["serde"] }
+tokio = { version = "1.40", features = ["sync", "rt"] }
```
Comment (NyxCode) [inline ts-rs/Cargo.toml:44]: Can we chose a somewhat older version of tokio? I think these types already existed in 1.0.

**Review by NyxCode — APPROVED** [review]
thanks! Happy to merge this once we double-check if we can't depend on an older tokio version.

File: ts-rs/Cargo.toml [hunk 2]
```diff
@@ -54,10 +56,10 @@ bson = { version = "2", optional = true }
...
 semver = { version = "1", optional = true }
-smol_str = { version = "0.3", optional = true  }
+smol_str = { version = "0.3", optional = true }
 thiserror = "1"
 indexmap = { version = "2", optional = true }
 ordered-float = { version = ">= 3, < 5", optional = true }
 serde_json = { version = "1", optional = true }
 lazy_static = { version = "1", default-features = false }
-
+tokio = { version = "1.40", optional = true }
```
Comment (NyxCode) [inline ts-rs/Cargo.toml:65]: Can we support an older version of tokio? I think even 1.0 should be fine?

Comment (joriskleiber) [reply ts-rs/Cargo.toml]: Yes, I updated dependency.

**Pull request merged**